packages/react-components/react-field/src/contexts/useFieldControlProps.ts

```diff
@@ -0,0 +1,99 @@
+import type { FieldContextValue, FieldControlProps } from '../Field';
+import { useFieldContext_unstable } from './FieldContext';
+
+/**
+ * Options for `useFieldControlProps_unstable`.
+ */
+export type FieldControlPropsOptions = {
+  /**
+   * Skips setting aria-labelledby on the control if the label.htmlFor refers to the control.
+   *
+   * This should be used with controls that can be the target of a label's htmlFor (e.g. input, textarea).
+   */
+  supportsLabelFor?: boolean;
+
+  /**
+   * True if the control has a size prop that matches Field's size prop.
+   */
+  supportsSize?: boolean;
+};
```

Comment:
Should there be a similar option for the `required` prop?

Reply:
👍 Added `supportsRequired` to use `required` instead of `aria-required` for controls that support it.